### PR TITLE
refactor: Use booleanAttribute transformer for boolean inputs

### LINKED ISSUE #13381: Utilize the booleanAttribute transformation
## Is your feature request related to a problem? Please describe.

Please update all of your boolean inputs to use the booleanAttribute transformation.

## Describe the solution you'd like

Instead of `@Input() filterable = true` use the new `@Input({transform: booleanAttribute}) filter = true`.

### PR DESCRIPTION
**Pull request by rkaraivanov**
Closes #13381   

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/directives/button/button.directive.ts [hunk 1]
```diff
@@ -230,30 +224,16 @@ export class IgxButtonDirective extends DisplayDensityBase {
...
-     * @example
-     * ```typescript
-     * const disabled = this.button.disabled;
-     * ```
-     */
+      * Enables/disables the button.
+      *
+      * @example
+      * ```html
+      * <button igxButton= "fab" [disabled]="true"></button>
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/directives/button/button.directive.ts:231]:
```suggestion
      * <button igxButton="fab" disabled></button>
```

File: projects/igniteui-angular/src/lib/directives/input/input.directive.ts [hunk 2]
```diff
@@ -63,8 +64,6 @@ export enum IgxInputState {
     standalone: true
 })
 export class IgxInputDirective implements AfterViewInit, OnDestroy {
-    private static ngAcceptInputType_required: boolean | '';
-    private static ngAcceptInputType_disabled: boolean | '';
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/directives/input/input.directive.ts:67]: Is this no longer needed so I can use the attribute without a value?

**Review by kdinev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/directives/input/input.directive.ts (hunk 2 above)
Comment (rkaraivanov) [reply projects/igniteui-angular/src/lib/directives/input/input.directive.ts:67]: Yes.

The changes to the boolean @Input properties (at least the simple ones) follow the pattern:

**Disabled state**

```html
<igx-switch disabled></igx-switch>
<igx-switch disabled=""></igx-switch>
<igx-switch disabled="true"></igx-switch>
```

**Enabled state**

```html
<igx-switch></igx-switch>
<igx-switch disabled="false"></igx-switch>
```

**Review by kdinev — DISMISSED** [review]

**Comment by kdinev** [comment]
@teodosiah Can you have someone go over the changes in the PR and do some smoke testing to make sure nothing is broken?

**Comment by ddaribo** [comment]
So far I found the following boolean input properties not using the `booleanAttribute` transformation. Not sure if they would be skipped intentionally, though:
1. `grid-editing-actions.component.ts` - [addChild](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/action-strip/grid-actions/grid-editing-actions.component.ts#L91)
2. `expandable-cell.component.ts` - [expanded](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/grids/grid/expandable-cell.component.ts#L50)
3. `groupby-row.component.ts` - [isFocused](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/grids/grid/groupby-row.component.ts#L103), [hideGroupRowSelectors](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/grids/grid/groupby-row.component.ts#L55), [rowDraggable](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/grids/grid/groupby-row.component.ts#L61)
4. `tree-cell.component.ts` - [showIndicator](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/grids/tree-grid/tree-cell.component.ts#L80), [isLoading](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/grids/tree-grid/tree-cell.component.ts#L86)
5. `switch.component.ts` - [disableRipple](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/switch/switch.component.ts#L187)
6. `years-view.component.ts` - [formatView](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/calendar/years-view/years-view.component.ts#L50)
7. `thumb-label.component.ts` - [continuous](https://github.com/IgniteUI/igniteui-angular/blob/rkaraivanov/input-transformers/projects/igniteui-angular/src/lib/slider/label/thumb-label.component.ts#L29)

Would also appreciate those directions for further smoke testing :)

**Comment by rkaraivanov** [comment]
@ddaribo 

Good catch! 
Everything that is an input property, and it is not `hidden/internal`, should be covered by the transform.
The rest are internal to the end user, so we've intentionally left them as there are now.

As for testing guidelines, I suggest following the patterns in https://github.com/IgniteUI/igniteui-angular/pull/13555#discussion_r1363705723 and checking if there are correctly applied in the samples/demos.

**Comment by ddaribo** [comment]
For the below properties, no errors are thrown when setting them via the usual property binding in the html template, i.e. `[propName]=”booleanValue”`, however the corresponding errors are thrown when using the syntax allowed by the booleanAttribute transformer, i.e. `propName` / `propName="booleanValue"` / `propName=""`.

1.	`IgxAccordion`: `singleBranchExpand` / `singleBranchExpand="true"` / `singleBranchExpand=""`

```
core.mjs:10592 ERROR Error: ASSERTION ERROR: Should be run in update mode [Expected=> false == true <=Actual]
    at throwError (core.mjs:328:11)
    at assertEqual (core.mjs:279:9)
    at refreshView (core.mjs:13468:18)
    at detectChangesInternal (core.mjs:13420:9)
    at ViewRef$1.detectChanges (core.mjs:13897:9)
    at IgxAccordionComponent.collapseAllExceptLast (accordion.component.ts:284:18)
    at set singleBranchExpand [as singleBranchExpand] (accordion.component.ts:128:18)
    at writeToDirectiveInput (core.mjs:12872:34)
    at setInputsFromAttrs (core.mjs:12853:13)
    at instantiateAllDirectives (core.mjs:12615:13)
```
dev sample: src\app\accordion\accordion.sample.html

2. `IgxColumnComponent`: setting `hasSummary` / `hasSummary ="true"` / `hasSummary =""` /  `hasSummary ="false"`

```
core.mjs:10592 ERROR Error: Uncaught (in promise): TypeError: Cannot read properties of undefined (reading 'summaryPipeTrigger')
TypeError: Cannot read properties of undefined (reading 'summaryPipeTrigger')
    at IgxGridSummaryService.resetSummaryHeight (grid-summary.service.ts:134:14)
    at set hasSummary (column.component.ts:317:38)
    at propDesc.set (watch-changes.ts:54:28)
    at propDesc.set (watch-changes.ts:83:32)
    at writeToDirectiveInput (core.mjs:12872:34)
    at setInputsFromAttrs (core.mjs:12853:13)
    at instantiateAllDirectives (core.mjs:12615:13)
    at createDirectivesInstances (core.mjs:12028:5)
    at ɵɵelementStart (core.mjs:16319:9)
    at GridSummaryComponent_Template (template.html:34:9)
    //....
```

dev sample: src\app\grid-summaries\grid-summaries.sample.html

3. `IgxColumnComponent`: setting `hidden` / `hidden="true"` / `hidden=""` - same error as # 2.

dev sample: any grid, i.e. src\app\grid\grid.sample.html

5. `IgxDialogComponent`: setting `isOpen` / `isOpen="true"` / `isOpen=""` / `isOpen="false"`

```
core.mjs:10194 ERROR Error: Uncaught (in promise): TypeError: Cannot read properties of undefined (reading 'collapsed')
TypeError: Cannot read properties of undefined (reading 'collapsed')
    at get isOpen [as isOpen] (igniteui-angular.mjs:40896:32)
    at set isOpen [as isOpen] (igniteui-angular.mjs:40899:28)
    at writeToDirectiveInput (core.mjs:11858:34)
    at setInputsFromAttrs (core.mjs:11839:13)
    at instantiateAllDirectives (core.mjs:11601:13)
    at createDirectivesInstances (core.mjs:11014:5)
    at Module.ɵɵelementStart (core.mjs:15319:9)
    at DialogSample1Component_Template (dialog-sample-1.component.html:3:1)
    //..
```
dev sample: src\app\dialog\dialog.sample.html

6. `IgxStep`: setting `active` / `active="true"` / `active=""`

```
core.mjs:10194 ERROR Error: ASSERTION ERROR: Should be run in update mode [Expected=> false == true <=Actual]
    at throwError (core.mjs:328:11)
    at assertEqual (core.mjs:279:9)
    at refreshView (core.mjs:12474:18)
    at detectChangesInternal (core.mjs:12429:9)
    at ViewRef$1.detectChanges (core.mjs:12903:9)
    at IgxStepperService.expandThroughApi (igniteui-angular.mjs:88503:29)
    at set active [as active] (igniteui-angular.mjs:88918:33)
    at writeToDirectiveInput (core.mjs:11858:34)
    at setInputsFromAttrs (core.mjs:11839:13)
    at instantiateAllDirectives (core.mjs:11601:13)
```
dev sample: src\app\stepper\stepper.sample.html

7. `IgxTreeGridGroupByAreaComponent`: setting `hideGroupedColumns` / `hideGroupedColumns="true"` / `hideGroupedColumns=""` / `hideGroupedColumns="false"`

```
core.mjs:10194 ERROR TypeError: Cannot read properties of undefined (reading 'columns')
    at set hideGroupedColumns [as hideGroupedColumns] (igniteui-angular.mjs:78820:23)
    at writeToDirectiveInput (core.mjs:11858:34)
    at setInputsFromAttrs (core.mjs:11839:13)
    at instantiateAllDirectives (core.mjs:11601:13)
    at createDirectivesInstances (core.mjs:11014:5)
    at ɵɵelementStart (core.mjs:15319:9)
    at Module.ɵɵelement (core.mjs:15377:5)
    at TreeGridSortingSampleComponent_Template (tree-grid-sorting-sample.component.html:26:9)
    at executeTemplate (core.mjs:10971:17)
    at renderView (core.mjs:12170:13)
```
dev sample: src\app\tree-grid-groupby\tree-grid-groupby.sample.html

8. The Angular Action Strip Paragraph Example (in [docs](https://www.infragistics.com/products/ignite-ui-angular/angular/components/action-strip)), from the igniteui-angular-samples [here](https://github.com/IgniteUI/igniteui-angular-samples/tree/master/src/app/menus/action-strip/action-strip-paragraph), has the `hidden` property of the `IgxActionStrip` initially set:
    `<igx-action-strip #actionStrip class="my-action-strip" [hidden]="true">`

and on `mouseenter`, `mouseleave` it is hidden/shown via the `hide()` and `show()` methods.

However, on replacing the hidden property bidning with the new syntax, i.e. `hidden="true"`, the action stip is no longer shown.

This might be due to the `hidden` attribute treated as the [HTMLElement's hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) ? Would be valid for all other boolean inputs named such.

8.1. The same is valid for the `IgxListItem` and its `hidden` property. The item is hidden even if `hidden="false"`.

**Comment by kdinev** [comment]
@rkaraivanov I think you should retarget this to `master` and relabel it `17.0.x`.

**Comment by rkaraivanov** [comment]
@ddaribo 
Can you check again after the last commit?

As for the **hidden** property for list item/action strip, it is indeed the native hidden attribute kicking in. I suppose the binding there should remain with the default Angular syntax, `[prop]="value"`

**Review by kdinev — APPROVED** [review]

**Pull request merged**